src/prefect/task_server.py

```diff
@@ -125,14 +124,25 @@ async def _submit_pending_task_runs(self, task_run_response):
             self._logger.info(repr(task_run.state))
 
             # The ID of the parameters for this run are stored in the Scheduled state's
-            # data value. If the data value is None, then the task run was created with
-            # no/empty parameters
+            # state_details. If there is no parameters_id, then the task was created
+            # without parameters.
             parameters = {}
-            if isinstance(task_run.state.data, LiteralResult):
-                parameter_id = UUID(task_run.state.data.value)
+            if hasattr(task_run.state.state_details, "task_parameters_id"):
+                parameters_id = task_run.state.state_details.task_parameters_id
                 task.persist_result = True
                 factory = await ResultFactory.from_task(task)
-                parameters = await factory.read_parameters(parameter_id)
+                try:
+                    parameters = await factory.read_parameters(parameters_id)
```

Comment:
We probably don't want this exception handler, I was just trying to cut down on the error output while the task engine couldn't find the files in storage.